src/app/time-period.ts

```diff
@@ -0,0 +1,103 @@
+export type Month = 0 | 1 | 2 | 3 | 4 | 5 | 6 | 7 | 8 | 9 | 10 | 11;
```

Comment:
Won't it be confusing to have the months be numbered starting at 0?

Reply:
Oh, sure, I see what you mean :) I suppose one-based indexing wouldn't be super useful here; there's no need for us be religious about starting from zero.

I'd be happy to change that sometime this weekend!

Reply:
Would you like the quarters to be one-indexed as well?

Reply:
I don't have a strong preference on this one. The other one felt like it was fighting a fairly standard way of referring to dates, but for the parts of the month I'm less concerned. What do others think?

Reply:
We have months zero-indexed because it makes the math easier and allows us to get the name of the month from an array eaisly.

Reply:
I mean, I suppose, but is it worth making the math easier if it makes the API less pretty?

Reply:
It might make sense for us, the implementors, to take on the burden of less-easy-math so that the users of the class have a better experience. ¯\\\_(ツ)\_/¯

Reply:
(Relevant: <https://xkcd.com/163/>)